sstables/compaction.cc

```diff
@@ -511,7 +519,22 @@ class compaction {
         report_start(formatted_msg);
 
         _compacting = std::move(ssts);
-        return make_sstable_reader();
+
+        auto& cs = _cf.get_compaction_strategy();
+        auto consumer = cs.make_interposer_consumer(_ms_metadata,
```

Comment:
BTW, the goal is to make TWCS major compaction time-window aware, but all other compaction types for TWCS will be made time-window aware too, like resharding, cleanup and regular compaction. The overhead of TWCS interposer consumer seems low, so I think this is okay

Reply:
Other compaction strategies' interposer may not be so efficient, so maybe it's better to have a separate virtual function for major compaction.

Reply:
It is definitely and absolutely the goal that the other compactions will create TWCS-compliant SSTables. Why shouldn't we?

Cleanup is not altered because the source is a single SSTable. If we assume the original SSTable is well behaved, it will result in a single SSTable.

For the others, like resharding, import, or whatever we are doing, why should we ever generate an SSTable that break the invariant of TWCS?

So the fact that this is quite central is by design.

Other strategies may have more expensive interposers, but currently they have no interposer at all and it is not clear they will ever have.